wirecardpaymentgateway/helper/OrderStateTransferObject.php

```
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Shop System Extensions:
+ * - Terms of Use can be found at:
+ * https://github.com/wirecard/prestashop-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/prestashop-ee/blob/master/LICENSE
+ */
+
+namespace WirecardEE\Prestashop\Helper;
+
+use Wirecard\ExtensionOrderStateModule\Domain\Contract\InputDataTransferObject;
+use Wirecard\ExtensionOrderStateModule\Domain\Entity\Constant;
+
+/**
+ * Class OrderStateTransferObject
+ * @package WirecardEE\Prestashop\Helper
+ * @since 2.10.0
+ */
+class OrderStateTransferObject implements InputDataTransferObject
+{
+    const FIELD_PROCESS_TYPE = "process-type";
+    const FIELD_TRANSACTION_TYPE = "transaction-type";
+    const FIELD_TRANSACTION_STATE = "transaction-state";
+    const FIELD_CURRENT_ORDER_STATE = "current-order-state";
+
+    /**
+     * @var string
+     */
+    private $processType = "";
+
+    /**
+     * @var string
+     */
+    private $transactionType = "";
+
+    /**
+     * @var string
+     */
+    private $transactionState = "";
+
+    /**
+     * @var int
+     */
+    private $currentOrderState = 0;
+
+    /**
+     * OrderStateTransferObject constructor.
+     * @param $currentOrderState
+     * @param $processType
+     * @param array $transactionResponse
+     */
+    public function __construct($currentOrderState, $processType, array $transactionResponse)
+    {
+
+        $this->setCurrentOrderState($currentOrderState);
```

Check the class invariants and throw exceptions. You don't want to transfer corrupt data to another system, thus infecting the other system as well. Fail early.

---

👍 But the question should we implement this checks in DTO object?